hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MembershipManager.java

```diff
@@ -112,7 +125,25 @@
         this.nodeEngine = node.getNodeEngine();
         this.logger = node.getLogger(getClass());
 
-        mastershipClaimTimeoutSeconds = node.getProperties().getInteger(MASTERSHIP_CLAIM_TIMEOUT_SECONDS);
+        this.mastershipClaimTimeoutSeconds = node.getProperties().getInteger(MASTERSHIP_CLAIM_TIMEOUT_SECONDS);
+
+        int partialDisconnectionResolutionHeartbeatCount = node.getProperties().getInteger(
+                PARTIAL_MEMBER_DISCONNECTION_RESOLUTION_HEARTBEAT_COUNT);
+        this.partialDisconnectionDetectionEnabled = partialDisconnectionResolutionHeartbeatCount > 0;
+        long heartbeatIntervalMs = SECONDS.toMillis(node.getProperties().getInteger(HEARTBEAT_INTERVAL_SECONDS));
+        long partialDisconnectivityDetectionIntervalMs =
+                partialDisconnectionResolutionHeartbeatCount * heartbeatIntervalMs;
+        long heartbeatTimeoutMs = SECONDS.toMillis(node.getProperties().getInteger(MAX_NO_HEARTBEAT_SECONDS));
+
+        if (!partialDisconnectionDetectionEnabled) {
+            partialDisconnectivityDetectionIntervalMs = Long.MAX_VALUE;
+        } else if (heartbeatTimeoutMs < partialDisconnectivityDetectionIntervalMs) {
+            partialDisconnectivityDetectionIntervalMs = heartbeatTimeoutMs;
+        }
+
+        this.partialDisconnectionHandler = new PartialDisconnectionHandler(partialDisconnectivityDetectionIntervalMs,
```

Comment:
Can we encapsulate these prop readings into `PartialDisconnectionHandler` to increase isolation?